### PR TITLE
Query History: Count using SQL, not post query

### LINKED ISSUE #50306: Query history: Test and improve indices and/or SQL queries
Spike to look at the SQL and see what kind of optimizations to the SQL we can do. 

Also use this time to investigate using K6 with this.

**What happened**:
In https://github.com/grafana/grafana/issues/44327 we have migrated query history to backend. It would be useful to do tests/improvements for indices and/or sql queries.

### PR DESCRIPTION
**Pull request by gelicia**
**What is this feature?**

To get the number of queries returned, the logic would query for all records, but return 1 for each record. In Go, it would then count how many records were returned. SQL has the ability to send a count of number of records returned, so we can use that.

This is to follow through on this comment https://github.com/grafana/grafana/pull/49217#discussion_r878769826

**Why do we need this feature?**

It is slightly more efficient.

**Which issue(s) does this PR fix?**:

Fixes #50306

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/queryhistory/database.go [hunk 1]
```diff
@@ -87,9 +87,9 @@ func (s QueryHistoryService) searchQueries(ctx context.Context, user *user.Signe
 		countBuilder := db.SQLBuilder{}
 		countBuilder.Write(`SELECT
 		`)
-		writeStarredSQL(query, s.store, &countBuilder)
+		writeStarredSQL(query, s.store, &countBuilder, true)
 		writeFiltersSQL(query, user, s.store, &countBuilder)
-		err = session.SQL(countBuilder.GetSQLString(), countBuilder.GetParams()...).Find(&allQueries)
+		_, err = session.SQL(countBuilder.GetSQLString(), countBuilder.GetParams()...).Get(&totalCount)
```
Comment (ivanahuckova) [inline pkg/services/queryhistory/database.go:92]: 🚀 🚀 🚀

Comment (ivanahuckova) [reply pkg/services/queryhistory/database.go:92]: This looks much more efficient.

Comment (ifrost) [reply pkg/services/queryhistory/database.go:92]: 👍 Is there a way to add a test for it? Out of curiosity, did you do any benchmarks?

Comment (gelicia) [reply pkg/services/queryhistory/database.go:92]: @ifrost Looking at the tests in `queryhistory_search_test.go`, it already looks fairly comprehensive (testing if it wants the count aggregation or not, and if we should return only starred results or not) and will test these code changes coming from the search handler. This function is the only one that uses `writeStarredSQL`

**Review by ifrost — APPROVED** [review]

**Pull request merged**